crates/solver-service/src/apis/order.rs

```diff
@@ -0,0 +1,140 @@
+//! OIF Solver Order API Implementation
+//!
+//! This module implements the order endpoint for the OIF Solver API, providing
+//! order retrieval functionality for cross-chain intents. Users can query the
+//! status and details of their submitted orders using the order ID.
+
+use axum::extract::Path;
+use solver_core::SolverEngine;
+use solver_storage;
+use solver_types::{
+	AssetAmount, DetailedIntentStatus, GetOrderError, GetOrderResponse, OrderResponse,
+	SettlementType,
+};
+use tracing::info;
+use uuid::Uuid;
+
+/// Handles GET /order/{id} requests.
+///
+/// This endpoint retrieves order details by ID, providing status information
+/// and execution details for cross-chain intent orders.
+pub async fn get_order_by_id(
+	Path(id): Path<String>,
+	_solver: &SolverEngine,
+) -> Result<GetOrderResponse, GetOrderError> {
+	info!("Retrieving order with ID: {}", id);
+
+	let order = process_order_request(&id, _solver).await?;
+
+	Ok(GetOrderResponse { order })
+}
+
+/// Processes an order retrieval request.
+async fn process_order_request(
+	order_id: &str,
+	solver: &SolverEngine,
+) -> Result<OrderResponse, GetOrderError> {
+	// Validate order ID format
+	validate_order_id(order_id)?;
+
+	// Try to retrieve the order from storage
+	match solver
+		.storage()
+		.retrieve::<solver_types::Order>("orders", order_id)
+		.await
+	{
+		Ok(order) => {
+			// Order found in storage, convert to OrderResponse
+			convert_order_to_response(order).await
+		}
+		Err(solver_storage::StorageError::NotFound) => {
+			// Order not found in storage
+			Err(GetOrderError::NotFound(format!(
+				"Order not found: {}",
+				order_id
+			)))
+		}
+		Err(e) => {
+			// Other storage error
+			Err(GetOrderError::Internal(format!("Storage error: {}", e)))
+		}
+	}
+}
+
+/// Validates the order ID format.
+fn validate_order_id(order_id: &str) -> Result<(), GetOrderError> {
+	// Check if it's a valid UUID format
+	if Uuid::parse_str(order_id).is_err() {
+		return Err(GetOrderError::InvalidId(format!(
+			"Order ID must be a valid UUID: {}",
+			order_id
+		)));
+	}
+
+	Ok(())
+}
+
+/// Converts a storage Order to an API OrderResponse.
+async fn convert_order_to_response(
+	order: solver_types::Order,
+) -> Result<OrderResponse, GetOrderError> {
+	// Extract data from the order's JSON data field
+	// This assumes the order.data contains the necessary fields for the API response
+	let input_amount = order
+		.data
+		.get("inputAmount")
+		.and_then(|v| serde_json::from_value::<AssetAmount>(v.clone()).ok())
+		.unwrap_or_else(|| AssetAmount {
+			asset: "0x0000000000000000000000000000000000000000".to_string(),
+			amount: alloy_primitives::U256::ZERO,
```

Comment:
I don't think we should be defaulting here. We need to gracefully handle them as errors. This goes for all the unwraps in this file..